pkgs/by-name/ed/edcas-client/package.nix

```diff
@@ -0,0 +1,82 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, openssl
+, cmake
+, wayland
+, libX11, libXcursor, libXi, libXrandr
+, libxkbcommon, libxcb
+, libglvnd
+}:
+
+let
+  rpathLibs = [
+    libglvnd
+    libXcursor
+    libXi
+    libxkbcommon
+    libXrandr
+    libX11
+    wayland
+  ];
+in rustPlatform.buildRustPackage rec {
+  pname = "edcas-client";
+  version = "0.2.6";
+
+  src = fetchFromGitHub {
+    owner = "arne-fuchs";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-1IkaulUeTdAbLSaVmINaAMrfAK5SSGFpW2l04PPZylA=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  cargoSha256 = "sha256-zYgNPlKCQ3g5p9W2jdcX/O21tCmklJh3om1oEOfo0QE=";
+
+  buildInputs = [
+    libxcb
+    libX11
+    libxkbcommon
+    openssl
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    rustPlatform.bindgenHook
+  ];
+
+  postPatch = ''
+    substituteInPlace ./src/app/materials.rs --replace '/usr/share/edcas-client/' "$out/usr/share/"
+    substituteInPlace ./src/app/settings.rs --replace '/etc/edcas-client/' "$out/etc/"
```

Comment:
This is likely not required. Doubtful though, should wait for another reviewer.

Reply:
The upstream package does the path handling manually so they have to be patched for the program to not crash.